### PR TITLE
Publish txs with min-relay-fee met

### LINKED ISSUE #1686: Check mempool min-relay-fee for each funding tx
We currently update our internal `mempool-min-relay-fee` once per block, but it actually changes whenever the mempool changes, so it may be obsolete when we try to publish a funding tx.

In `BitcoinCoreWallet`, in `fundTransaction`, we should start by querying `getmempoolinfo` and raise the feerate if necessary.

### PR DESCRIPTION
**Pull request by t-bast**
When our mempool is full, its min-relay-fee may be constantly changing.
To ensure our txs can be published, we need to check the min-relay-fee when we fund the transaction, and raise it if necessary.

Fixes #1686

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWallet.scala [hunk 1]
```diff
@@ -42,12 +42,20 @@ class BitcoinCoreWallet(rpcClient: BitcoinJsonRPCClient)(implicit ec: ExecutionC
...
-      FundTransactionResponse(Transaction.read(hex), changepos.intValue, toSatoshi(fee))
+    val requestedFeeRatePerKB = FeeratePerKB(feeRatePerKw)
+    rpcClient.invoke("getmempoolinfo").map(json => json \ "mempoolminfee" match {
+      case JDecimal(feerate) => FeeratePerKB(Btc(feerate).toSatoshi).max(requestedFeeRatePerKB)
+      case JInt(feerate) => FeeratePerKB(Btc(feerate.toLong).toSatoshi).max(requestedFeeRatePerKB)
+      case other =>
+        logger.warn(s"cannot retrieve mempool minimum fee: $other")
+        requestedFeeRatePerKB
+    }).flatMap(feeRatePerKB => {
+      rpcClient.invoke("fundrawtransaction", hex, Options(lockUnspents, BigDecimal(feeRatePerKB.toLong).bigDecimal.scaleByPowerOfTen(-8))).map(json => {
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWallet.scala:53]: Nit (there is also an unused import)
```suggestion
      rpcClient.invoke("fundrawtransaction", hex, Options(lockUnspents, feeRatePerKB.toLong.bigDecimal.scaleByPowerOfTen(-8))).map(json => {
```

Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWallet.scala:52]: That doesn't compile... :thinking:

Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWallet.scala:52]: Requires `import scala.math.BigDecimal.long2bigDecimal`.

Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWallet.scala]: Good catch, my currently broken IntelliJ IDEA didn't suggest that :)
Done in https://github.com/ACINQ/eclair/pull/1687/commits/ec55eff5058166a65afb4760e7fbe152d76625be

**Review by pm47 — APPROVED** [review]

**Pull request merged**